File: altanto-backend-main/src/main/java/com/course/altanto/entity/Product.java
package com.course.altanto.entity;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToMany;

import org.hibernate.annotations.GenericGenerator;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * @author gabriel.juarez
 *
 */

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity
public class Product implements Serializable {

	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(generator = "uuid2")
	@GenericGenerator(name = "uuid2", strategy = "uuid2")
	private String id;

	@Column(nullable = false)
	private String code;

	@Column(nullable = false)
	private String name;

	@Column(nullable = false)
	private String shortDescription;

	private int cantd;
	
	private String size;
	
	@Column(nullable = false)
	private double price;
	
	private int discount;
	
	private String category;
	
	private double rating;
	
	private String typeGarment;
	
	@ManyToMany(targetEntity = File.class,cascade = CascadeType.ALL)
	private List<File> images;
	
	private Date createdAt;
	
	private Date editAt;
	
	
}

File: dashboard_angular-main/src/app/models/Product.ts
    export class Product {
        id: string;
        code: string;
        name: string;
        shortDescription: string;
        cantd: any;
        size: string;
        price: any;
        createdAt: string;
        discount: any;
        category: string;
        rating: any;
        typeGarment: string;
        typeClothing : string;
        images: Image[];

        constructor() {
            this.id = '';
            this.code = '';
            this.name = '';
            this.shortDescription = '';
            this.cantd = 0;
            this.size = '';
            this.price = 0;
            this.createdAt = '';
            this.discount = 0;
            this.category = '';
            this.rating = 0;
            this.typeGarment = '';
            this.typeClothing = '';
            this.images = [];
        }
    }
 
    export class Image {
        id: string;
        routeFile: string;
        nameEntity: string;
        nameFile: string;

        constructor() {
            this.id = '';
            this.routeFile = '';
            this.nameEntity = '';
            this.nameFile = '';
        }
    }